codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -832,8 +846,9 @@ public void flowController(F flowController) {
         }
 
         private void checkNewStreamAllowed(int streamId) throws Http2Exception {
-            if (goAwaySent() || goAwayReceived()) {
-                throw connectionError(PROTOCOL_ERROR, "Cannot create a stream since the connection is going away");
+            if (goAwayReceived()) {
```

Comment:
@buchgr +1.  @nmittler - For this issue lets just focus on correctness w.r.t. the spec. We can always handle extensibility in another issue.